src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFileStream.cs

```diff
@@ -52,10 +53,32 @@ public IsolatedStorageFileStream(string path, FileMode mode, FileAccess access,
         {
         }
 
-        // If the isolated storage file is null, then we default to using a file
-        // that is scoped by user, appdomain, and assembly.
-        public IsolatedStorageFileStream(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, IsolatedStorageFile isf) :
-            base (path, mode, access, share)
+        public IsolatedStorageFileStream(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, IsolatedStorageFile isf)
+            : this(path, mode, access, share, bufferSize, InitializeFileStream(path, mode, access, share, bufferSize, isf))
+        {
+        }
+
+        // On NetFX FileStream has an internal no arg constructor that we utilize to provide the facade. We don't have access
+        // to internals in CoreFX so we'll do the next best thing and contort ourselves into the SafeFileHandle constructor.
+        // (A path constructor would try and create the requested file and give us two open handles.)
+        private IsolatedStorageFileStream(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, InitialiationData initializationData)
+            : base(new SafeFileHandle(initializationData.NestedStream.SafeFileHandle.DangerousGetHandle(), ownsHandle: false), access, bufferSize)
```

Comment:
Can you add a comment about why this DangerousGetHandle call is safe?  I assume it's because we know the only caller will be the other ctor that's creating the stream fresh and without handing it out to anyone else that might close the handle?